beluga/test/benchmark/benchmark_take_while_kld.cpp

```diff
@@ -93,22 +90,19 @@ void BM_AdaptiveResample(benchmark::State& state) {
 
   std::size_t min_samples = 0;
   std::size_t max_samples = container_size;
-  auto hasher = beluga::spatial_hash<State>{};
   double kld_epsilon = 0.05;
   double kld_z = 3.;
 
   for (auto _ : state) {
-    auto&& samples = ranges::views::generate(beluga::make_multinomial_sampler(
-                         beluga::views::states(container), beluga::views::weights(container), generator)) |
-                     beluga::views::take_while_kld(hasher, min_samples, max_samples, kld_epsilon, kld_z) |
-                     ranges::views::take(container_size) | ranges::views::transform([](const auto& state) {
-                       return Particle{state, 1.0};
-                     }) |
-                     ranges::views::common;
-    auto first = std::begin(new_container);
-    auto last = std::copy(std::begin(samples), std::end(samples), first);
-    state.counters["SampleSize"] = static_cast<double>(std::distance(first, last));
-    state.counters["Percentage"] = static_cast<double>(std::distance(first, last)) / static_cast<double>(max_samples);
+    auto samples =
+        container |              //
+        beluga::views::sample |  //
+        beluga::views::take_while_kld(beluga::spatial_hash<State>{}, min_samples, max_samples, kld_epsilon, kld_z);
```

Comment:
@nahueespinosa meta: would it make sense for `beluga::views::take_while_kld` to use `beluga::spatial_hash<State>{}` if none is provided? IIUC we should be able to deduce `State` from `range_value_t<Range>` for the corresponding overload.

Reply:
We can, but `spatial_hash` requires configuration parameters (i.e. resolution) that we would need to pass by value or provide sensible defaults. Both are reasonable. I'll sleep on it but if you have more thoughts on what API you'd like to see/use, that'd be great.